//Define the used version of proto
syntax = "proto3";

import "google/protobuf/empty.proto";
package data_client;

message DataSample {
  float pm1 = 1;
  float pm25 = 2;
  float pm10 = 3;
  float air_temperature = 4;
  float relative_humidity = 5;
  float precipitation = 6;
  float air_pressure = 7;
  float wind_speed = 8;
  float wind_direction = 9;
}

message ClientInitRequest {
  string iot_api = 1;
  string iot_token = 2;
  string met_id = 3;
}

//Define the service
service DataClient {
  rpc init_client(ClientInitRequest) returns (google.protobuf.Empty);
  rpc get_last_data(google.protobuf.Empty) returns (DataSample);
}